device/transport/amqp/src/amqp.ts

```diff
@@ -283,7 +287,12 @@ export class Amqp extends EventEmitter implements DeviceTransport {
                   /*Codes_SRS_NODE_DEVICE_AMQP_13_002: [ The connect method shall set the CA cert on the options object when calling the underlying connection object's connect method if it was supplied. ]*/
                   if (this._options && this._options.ca) {
                     config.sslOptions = config.sslOptions || {};
-                    config.sslOptions.caFile = this._options.ca;
+                    /*Codes_SRS_NODE_DEVICE_AMQP_06_012: [The `connect` method shall first test if the `ca` property is the name of an already existent file.  If so, it will attempt to read that file as a pem into a string value and pass the string to config object `ca` property.  Otherwise, it is assumed to be a pem string.] */
+                    if (fs.existsSync(this._options.ca)) {
```

Comment:
>if (fs.existsSync(this._options.ca)) { [](start = 20, length = 38)

It seems dangerous to pass an arbitrary string to a filesystem function.  Would it make more sense to check if the string begins with the -----BEGIN... thing?

Reply:
that's a good point. didn't think about that when I suggested this. there are security implications with even checking if the file exists too. Maybe we should run this by a security person.  

---
In reply to: [214414255](https://github.com/Azure/azure-iot-sdk-node/pull/354#discussion_r214414255) [](ancestors = 214414255)

Reply:
I actually thought about this quite a bit.  My assumption was that the node file api would be sufficiently careful about this.  I was actually made even more comfortable about this when I looked to see whether the api would throw.  It doesn't.

---
In reply to: [214525406](https://github.com/Azure/azure-iot-sdk-node/pull/354#discussion_r214525406) [](ancestors = 214525406,214414255)